### ISSUE TITLE
conj(::Any) should be deprecated

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaLang/julia#26248]
`conj(::Any)` should be deprecated, by the same logic as `transpose(::Any)`:

```julia
julia> struct Foo end

julia> conj(Foo())
Foo()

julia> transpose(Foo())
ERROR: MethodError: no method matching transpose(::Foo)
Closest candidates are:
  transpose(::Missing) at missing.jl:75
  transpose(::Number) at number.jl:174
  transpose(::LinearAlgebra.Transpose) at /Users/solver/Projects/julia7-branch/usr/share/julia/site/v0.7/LinearAlgebra/src/adjtrans.jl:105
```

**Comment by Sacha0** [comment]
Agreed. This method was the root cause of the recent issue with conjugation of `I` failing. Best!

**Comment by JeffBezanson** [comment]
Triage accepts. @dlfivefifty are you available to work on this?

**Comment by dlfivefifty** [comment]
Sure.

**Comment by JeffBezanson** [comment]
Fixed by #26288.
